src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs

```diff
@@ -0,0 +1,154 @@
+            _logger = loggerProvider.CreateLogger(nameof(InvocationHandler));
+        }
+
+        public async Task<InvocationResponse> InvokeAsync(InvocationRequest request)
+        {
+            FunctionContext? context = null;
+            InvocationResponse response = new()
+            {
+                InvocationId = request.InvocationId,
+                Result       = new StatusResult()
+            };
+
+            try
+            {
+                var invocation = new GrpcFunctionInvocation(request);
+                var cancellationTokenSource = new CancellationTokenSource();
+
+                if (!_inflightInvocations.TryAdd(request.InvocationId, cancellationTokenSource))
+                {
+                    // Dispose the CancellationTokenSource if we cannot keep track of it
+                    cancellationTokenSource?.Dispose();
+                }
+
+                IInvocationFeatures invocationFeatures = _invocationFeaturesFactory.Create();
+                invocationFeatures.Set<FunctionInvocation>(invocation);
+                invocationFeatures.Set<IExecutionRetryFeature>(invocation);
+
+                var token = cancellationTokenSource?.Token ?? CancellationToken.None;
+                context = _application.CreateContext(invocationFeatures, token);
+                invocationFeatures.Set<IFunctionBindingsFeature>(new GrpcFunctionBindingsFeature(context, request, _outputBindingsInfoProvider));
+
+                if (_inputConversionFeatureProvider.TryCreate(typeof(DefaultInputConversionFeature), out var conversion))
+                {
+                    invocationFeatures.Set<IInputConversionFeature>(conversion!);
+                }
+
+                await _application.InvokeFunctionAsync(context);
+
+                var functionBindings = context.GetBindings();
+
+                foreach (var binding in functionBindings.OutputBindingData)
+                {
+                    var parameterBinding = new ParameterBinding
+                    {
+                        Name = binding.Key
+                    };
+
+                    if (binding.Value is not null)
+                    {
+                        parameterBinding.Data = await binding.Value.ToRpcAsync(_serializer);
+                    }
+
+                    response.OutputData.Add(parameterBinding);
+                }
+
+                if (functionBindings.InvocationResult != null)
+                {
+                    TypedData? returnVal = await functionBindings.InvocationResult.ToRpcAsync(_serializer);
+                    response.ReturnValue = returnVal;
+                }
+
+                response.Result.Status = StatusResult.Types.Status.Success;
+            }
+            catch (Exception ex)
+            {
+                response.Result.Exception = ex.ToRpcException();
+                response.Result.Status = StatusResult.Types.Status.Failure;
+
+                if (ex.InnerException is TaskCanceledException)
```

Comment:
Do we need to also check for `OperationCancelledException`?

```suggestion
                if (ex.InnerException is TaskCanceledException or OperationCancelledException)
```